File: Javascript_sol/Arrays_hashing/Leetcode_9.js
/**
 * @param {number} x
 * @return {boolean}
 */

// Convert the integer to a string and initiate a loop
// with 2 pointers till the middle
// Time Complexity: O(2n) ; Space Complexity: O(n)
var isPalindrome = function(x) {
    const y = x.toString()
    const size = y.length;

    for (i=0; i<=Math.ceil(size/2); i++) {
        if (y[i] != y[size-1-i]) {
            return false;
        }
    }
    return true;
};

File: Python_sol/Arrays_hashing/Leetcode_9.py

#  Convert the integer to a string and initiate a loop
#  with 2 pointers till the middle with
#  Time Complexity: O(n+n/2) ; Space Complexity: O(n)
class Solution:
    def isPalindrome(self, x: int) -> bool:
        y = str(x)
        for i in range(int(len(y)/2)):
            if y[i] != y[-1-i]:
                return 0
        return 1